lambda/matching.ml

```diff
@@ -3431,7 +3439,52 @@ and combine_handlers ~scopes repr partial ctx (v, str, arg) first_match rem =
          repr)
       partial ctx first_match rem
   in
-  (bind_check str v arg lam, total)
+  let jumps =
```

Comment:
My intuition (but I may be out of the loop here) is that the context information can be both "locally correct" and "globally incorrect". Inside the scope of the binding `bind_check str v arg lam`, all accesses to the argument are made through the (immutable) variable `v`, so the context information is locally correct. In particular, if we use it to compute a "failure case", that computation will be correct. On the other hand, what is true of `v` is not necessarily true of `arg` in general when that argument expression is impure, so returning this context information to our caller may be incorrect.

(It is not obvious to see the connection between `v`, `arg` on one hand and the code in `first_match` and `rem` on the other, because that connection is made in the *caller* of `combine_handlers`. I needed to look at `compile_match_nonempty` to convince myself again that this is how things work.)